src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFJobManager.groovy

```diff
@@ -326,7 +326,7 @@ class LSFJobManager extends AbstractLSFJobManager {
     private File getBjobsFile(String s, BEJobID jobID, String type) {
         if (!s) {
             return null
-        } else if (executionService.execute("stat -c %F ${BashUtils.strongQuote(s)}").firstLine == "directory") {
+        } else if (executionService.execute("LC_ALL=C stat -c %F ${BashUtils.strongQuote(s)} 2> /dev/null").firstLine == "directory") {
```

Comment:
Rename "type" to "typeFileSuffix".

If a "directory" is assumed by the caller in argument `s`, but the directory does not exist yet, the code will create a filename for a file (else-branch). This is probably not intended. Maybe add a second method `getBjobsDirectory` for the directory case, rather than this buggy logic.

Please add comment, that the file does not have to exist.

Reply:
Your changes are fine.

Reply:
The reason why it is done this way is that you can pass either a file or a directory as log path, and LSF doesn't return the actual path to the log file, but the path that was passed to it (unlike eg. PBS).

Because BE should abstract over this differences, we try to find out here the correct path to the _actual file_ that LSF creates.

Reply:
I would add a comment explaining this.

Reply:
Sorry, I explained it wrongly. The second branch with the stat command is taken, if the stat command succeeds and returns "directory" as file. However, if the stat command fails, the terminal else is taken, even though, the `s` may still be a directory name.

It should be checked, whether the stat command succeeded (which should AFAICR not be the case if the pathname does not point to a file/directory), and only upon success, the output "directory" should be checked.

If the stat fails, there are two options. Either a directory or file is referred to. I am not sure whether we can discriminate the two cases. Do we know them from the way we deal with LSF?